src/client/VZSidebar/Item.tsx

```diff
@@ -64,6 +64,19 @@ export const Item = ({
     [id, setHoveredItemId],
   );
 
+  const validateFileName = useCallback(
```

Comment:
Please refactor this so that there is only a single definition of `validateFileName` in the codebase. I see there is another definition of this over in `src/client/VZSidebar/CreateDirModal.tsx`. Please pull this function out into its own separate file, then call it from both places. Thanks!

Reply:
Something like this:

```
export const validateFileName = ({files}) => {
  // Returns true if file name is valid, false otherwise.
  return useCallback(
    (fileName: string) => {
      let valid;
      // General Character Check
      const regex =
        /^[a-zA-Z0-9](?:[a-zA-Z0-9 ./+=_-]*[a-zA-Z0-9])?$/;
      valid = regex.test(fileName);

      // Check for Duplicate Filename
      for (const key in files) {
        if (fileName + '/' === files[key].name) {
          valid = false;
        }
      }

      return valid;
    },
    [files],
  );
}
```

This is a "custom hook" in React parlance.